### PR TITLE
feat(site): move resources into the sidebar

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
https://github.com/coder/coder/assets/3165839/2cdfca6b-f122-4a71-8eff-fade960b0b65

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/11491

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            color: theme.palette.text.secondary,
+            lineHeight: "1.5",
+          }}
+        >
+          Your workspace build failed, so the necessary resources couldn&apos;t
+          be created.
+        </p>
+      )}
+      {resources.length === 0 &&
+        !failed &&
```
Comment (Parkreiner) [inline site/src/pages/WorkspacePage/ResourcesSidebar.tsx:41]: Is this condition strictly for handling loading states?

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          isActive={r.id === selected}
+          key={r.id}
+          css={styles.root}
+        >
+          <div
+            css={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              lineHeight: 0,
```
Comment (Parkreiner) [inline site/src/pages/WorkspacePage/ResourcesSidebar.tsx:59]: Do you want a lineHeight of 0, or a lineHeight of 1? I've had issues where a value of 0 broke a lot of text rendering and made text invisible the moment something accidentally overflowed to a new line

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          </div>
+        </SidebarItem>
+      ))}
+    </Sidebar>
+  );
+};
+
+export const ResourceSidebarItemSkeleton = () => {
+  return (
+    <div css={[styles.root, { pointerEvents: "none" }]}>
```
Comment (Parkreiner) [inline site/src/pages/WorkspacePage/ResourcesSidebar.tsx:87]: Does `pointerEvents` do anything here? Divs aren't interactive by default, and turning on the style rule would just make clicks on the div hit the parent directly, which wouldn't be _that_ different from the events firing on the div, and then bubbling up to the parent

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              lineHeight: 0,
+              width: 16,
+              height: 16,
+              padding: 2,
+            }}
+          >
+            <img
```
Comment (Parkreiner) [inline site/src/pages/WorkspacePage/ResourcesSidebar.tsx:65]: Can you add `role="presentation"` here?

File: site/src/pages/WorkspacePage/Workspace.tsx [hunk 5]
```diff
@@ -148,6 +145,29 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
...
+    const { set, value } = sidebarOption;
+    if (value === newOption) {
+      set("");
+    } else {
+      set(newOption);
+    }
+  };
+
+  const selectedResourceId = useTab("resources", "");
+  const resources = workspace.latest_build.resources.sort(
```
Comment (Parkreiner) [inline site/src/pages/WorkspacePage/Workspace.tsx:159]: Accidental mutation here

**Review by buenos-nachos — APPROVED** [review]
Looks good! Just had a few suggestions for improving some of the markup, and caught one tiny, React-specific bug

File: site/src/utils/workspace.tsx [hunk 6]
```diff
@@ -285,3 +285,23 @@ export const workspaceUpdatePolicy = (
   }
   return workspace.automatic_updates;
 };
+
+// These resources (i.e. docker_image, kubernetes_deployment) map to Terraform
+// resource types. These are the most used ones and are based on user usage.
+// We may want to update from time-to-time.
+const BUILT_IN_ICON_PATHS: Record<string, string> = {
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/utils/workspace.tsx:292]: love the idea

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/ResourcesSidebar.tsx:41]: Yes.

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/ResourcesSidebar.tsx:59]: I think 1 🤔 but let me check.

Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/ResourcesSidebar.tsx:59]: Ahh in this case, it is an image wrapper so it is ok to use 0... I'm going to replace those wrappers - we have a few in the codebase - with the new @aslilac component `ExternalIcon`

File: site/src/pages/WorkspacePage/Workspace.tsx (hunk 5 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/Workspace.tsx:159]: Not sure if I got it 🤔

File: site/src/pages/WorkspacePage/ResourcesSidebar.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/ResourcesSidebar.tsx:65]: Yeap.

File: site/src/pages/WorkspacePage/Workspace.tsx (hunk 5 above)
Comment (Parkreiner) [reply site/src/pages/WorkspacePage/Workspace.tsx:159]: So, the `.sort` method is mutating the underlying array, even though that array is globally available throughout the app through the query cache. That could cause bugs for any other components that need the data in its original form

I don't think the `.toSorted` method is supported enough to swap that in, but you should be able to rewrite this as
```ts
 const resources = [...workspace.latest_build.resources].sort()
```

**Pull request merged**